### ISSUE TITLE
Some fields namely password need to be hashed 

### ISSUE THREAD

**Issue by jrans** [dwyl/abase#52]
Use [bcrypt](https://www.npmjs.com/package/bcrypt) to hash fields before saving confidential information which should not be stored in db like password. 

Have functionality to hash any payload based on fields config but have put on shelf. #58 

Now will only hash "password" field before saying as our sole means of verification.

**Comment by nelsonic** [comment]
Passwords need to be _hashed_ when they are _inserted_ into the Database.
But they need to be in `plaintext` when you are _comparing_ to an _existing_ password that is _stored_ in the DB so hashing it prematurely and discarding the `plaintext` is kinda useless ... 😕 
see: https://github.com/ncb000gt/node.bcrypt.js/#to-check-a-password 

https://www.youtube.com/watch?v=1in5wAVOyIk ❓

**Comment by jrans** [comment]
@nelsonic yep realised, though maybe with asynchronous validation functions we could use the compare method of bcrypt with a db request could be useful to validate the login request with joi? See progress of [async extensions](https://github.com/hapijs/joi/issues/577) (could be also be used for checking username unique etc) but yes original proposal flawed!

**Comment by jrans** [comment]
Note columns in db storing hashes need to be of length [60char](http://stackoverflow.com/questions/5881169/what-column-type-length-should-i-use-for-storing-a-bcrypt-hashed-password-in-a-d)
